boot/espressif/port/esp_mcuboot.c

```diff
@@ -218,36 +237,55 @@ int flash_area_read(const struct flash_area *fa, uint32_t off, void *dst,
     return 0;
 }
 
-static bool aligned_flash_write(size_t dest_addr, const void *src, size_t size)
+static bool aligned_flash_write(size_t dest_addr, const void *src, size_t size, bool erase)
 {
 #ifdef CONFIG_SECURE_FLASH_ENC_ENABLED
```

Comment:
`esp_flash_encryption_enabled();` will return if encryption is enabled. Why the need of `#ifdef CONFIG_SECURE_FLASH_ENC_ENABLED`?